src/plugin/managers/lifecycle.spec.ts

```
@@ -0,0 +1,498 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest';
+import { LifecycleManager } from './lifecycle';
+import { PluginStatus } from '../types';
+
+// --- Mocks ---
+
+const mockRegisterPluginDynamically = vi.fn();
+vi.mock('../registry', () => ({
+  registerPluginDynamically: (...args: any[]) => mockRegisterPluginDynamically(...args),
```

ESLint

[error] Insert `⏎···`

(prettier/prettier)